benchmarks/src/main/scala/benchmarks/Benchmark.scala

```diff
@@ -59,30 +35,19 @@ abstract class Benchmark[T] {
 
   final def loop(iterations: Int): BenchmarkResult =
     try {
-      var min: Long        = Long.MaxValue
-      var max: Long        = Long.MinValue
       var success: Boolean = true
 
-      val t0 = System.nanoTime()
-      for (_ <- 1 to iterations) {
-        val start  = System.nanoTime()
-        val result = run()
-        val end    = System.nanoTime()
+      val times =
+        for (_ <- 1 to iterations) yield {
+          val start  = System.nanoTime()
+          val result = run()
+          val end    = System.nanoTime()
 
-        success = success && check(result)
-        val elapsed = end - start
-        if (elapsed > max) max = elapsed
-        if (elapsed < min) min = elapsed
-      }
-      val totalTime = System.nanoTime() - t0
-      val average   = totalTime / iterations
+          success = success && check(result)
+          end - start
+        }
```

Comment:
Lets do a while loop here and store the results into an array. Otherwise I'm afraid of accidental overheads of the wrapping code that we should avoid at all cost.